llama_stack/providers/tests/memory/test_vector_store.py

```diff
@@ -0,0 +1,68 @@
+import base64
+import mimetypes
+import os
+from pathlib import Path
+import pytest
+from llama_stack.providers.utils.memory.vector_store import content_from_doc
+from llama_stack.apis.memory.memory import MemoryBankDocument, URL
+from pytest_httpx import HTTPXMock
+
+DUMMY_PDF_PATH = Path(os.path.abspath(__file__)).parent / "fixtures" / "dummy.pdf"
+
+def read_file(file_path: str) -> bytes:
+    with open(file_path, "rb") as file:
+        return file.read()
+
+def data_url_from_file(file_path: str) -> str:
+    with open(file_path, "rb") as file:
+        file_content = file.read()
+
+    base64_content = base64.b64encode(file_content).decode("utf-8")
+    mime_type, _ = mimetypes.guess_type(file_path)
+
+    data_url = f"data:{mime_type};base64,{base64_content}"
+
+    return data_url
+
+# Requires pytest-httpx
+
+class TestVectorStore:
+    @pytest.mark.asyncio
+    async def test_returns_content_from_pdf_data_uri(self):
+        data_uri = data_url_from_file(DUMMY_PDF_PATH)
+        doc = MemoryBankDocument(
+            document_id="dummy",
+            content=data_uri,
+            mime_type="application/pdf",
+            metadata={},
+        )
+        content = await content_from_doc(doc)
+        assert content == "Dummy PDF file"
+    
+    @pytest.mark.asyncio
+    async def test_downloads_pdf_and_returns_content(self, httpx_mock: HTTPXMock):
+        url = "https://example.com/dummy.pdf"
+        httpx_mock.add_response(url=url, content=read_file(DUMMY_PDF_PATH))
```

Comment:
Not sure what you guys think about mocking httpx here. Other options are:

- Download from a third party url: https://www.w3.org/WAI/ER/tests/xhtml/testfiles/resources/pdf/dummy.pdf
- Host it somewhere that we control (maybe this repo?)

Lmk what your preferences are here